openAPI/core-back-internal.yaml

```diff
@@ -60,6 +60,21 @@ paths:
           Fn::Sub: arn:aws:apigateway:${AWS::Region}:states:action/StartSyncExecution
         passthroughBehavior: "when_no_match"
         requestTemplates:
+          application/json:
+            # This VTL template maps the request into the format required by the step function API
+            # Note that the lack of indentation is deliberate:
+            # the 'input' field must be condensed into a single line to be a valid JSON string
```

Comment:
I would welcome alternate approaches - the problematic constraint is that the `input` value is a JSON string containing a JSON object (not a raw JSON object), so must be a single line.

This is the best I could come up with that still allows us to write the different parts on different lines (so you can easily see all the properties being passed through) - you just can't do indentation or YAML won't strip the newlines.

Reply:
We could use a flow scalar, which means that new lines are removed, even if there's an indent. It's a little heavy on the escapes though. I've tested this and it seems to work.
```
          application/json:
            Fn::Sub: "{
              \"input\": \"{
                \\\"journey\\\":\\\"/journey/$util.escapeJavaScript($input.params('journeyStep'))\\\",
                \\\"journey\\\":\\\"/journey/$util.escapeJavaScript($input.params('journeyStep'))\\\",
                \\\"ipAddress\\\":\\\"$util.escapeJavaScript($input.params('ip-address'))\\\",
                \\\"featureSet\\\":\\\"$util.escapeJavaScript($input.params('feature-set'))\\\",
                \\\"ipvSessionId\\\":\\\"$util.escapeJavaScript($input.params('ipv-session-id'))\\\",
                \\\"clientOAuthSessionId\\\":\\\"$util.escapeJavaScript($input.params('client-session-id'))\\\"
              }\",
              \"stateMachineArn\": \"arn:aws:states:${AWS::Region}:${AWS::AccountId}:stateMachine:${JourneyEngineStepFunction.Name}\"
            }"
```

It's annoying that the argument to input params needs to be quoted, otherwise we could have done something like below. You can't use escapes in a single quoted scalar, and I can't figure out a way to make it work...
```
Fn::Sub: '{
  "input": "{
    \"journey\":\"/journey/$util.escapeJavaScript($input.params(journeyStep))\",
    \"journey\":\"/journey/$util.escapeJavaScript($input.params(journeyStep))\",
    \"ipAddress\":\"$util.escapeJavaScript($input.params(ip-address))\",
    \"featureSet\":\"$util.escapeJavaScript($input.params(feature-set))\",
    \"ipvSessionId\":\"$util.escapeJavaScript($input.params(ipv-session-id))\",
    \"clientOAuthSessionId\":\"$util.escapeJavaScript($input.params(client-session-id))\"
  }",
  "stateMachineArn": "arn:aws:states:${AWS::Region}:${AWS::AccountId}:stateMachine:${JourneyEngineStepFunction.Name}"
}'
```

Reply:
Hmm, I can't quite decide which I think is better!

Reply:
I'll appove and then you can decide if you want to change it!